mythril_core/src/logger.rs

```diff
@@ -1,22 +1,29 @@
 use crate::time;
+use crate::vga_logger::VgaWriter;
 
 use core::fmt;
 use core::fmt::Write;
 use spin::Mutex;
 
 static LOG_LOCK: Mutex<()> = Mutex::new(());
+static VGA_LOCK: Mutex<VgaWriter> = Mutex::new(VgaWriter::new());
 
-pub fn write_console(s: impl AsRef<str>) {
+pub fn write_console(s: &str) {
     let lock = LOG_LOCK.lock();
     unsafe { raw_write_console(s) };
-    drop(lock)
+    drop(lock);
 }
 
 // NOTE: the caller should hold `LOG_LOCK`
-pub unsafe fn raw_write_console(s: impl AsRef<str>) {
+pub unsafe fn raw_write_console(s: &str) {
+    // mirror console output to VGA
+    let mut vga_writer = VGA_LOCK.lock();
```

Comment:
I think the issue here is that if we panic while we hold the VGA lock, then the panic handler will try to acquire the lock again and we will deadlock